Index: Makefile.am
===================================================================
--- Makefile.am	(revision 28689)
+++ Makefile.am	(working copy)
@@ -1,5 +1,5 @@
 INCLUDES = -I$(top_srcdir)/src/include
-SUBDIRS  = contrib doc m4 src po pkgconfig
+SUBDIRS  = libltdl contrib doc m4 src po pkgconfig
 
 EXTRA_DIST = \
  ABOUT-NLS \
Index: configure.ac
===================================================================
--- configure.ac	(revision 28689)
+++ configure.ac	(working copy)
@@ -506,9 +506,6 @@
 AC_CHECK_HEADERS([fcntl.h math.h errno.h ctype.h limits.h stdio.h stdlib.h string.h unistd.h stdarg.h signal.h locale.h sys/stat.h sys/types.h],,AC_MSG_ERROR([Compiling GNUnet requires standard UNIX headers files]))
 
 
-# Check for libltdl header (#2999)
-AC_CHECK_HEADERS([ltdl.h],,AC_MSG_ERROR([Compiling GNUnet requires libltdl headers (try installing libltdl-dev)]))
-
 # Checks for headers that are only required on some systems or opional (and where we do NOT abort if they are not there)
 AC_CHECK_HEADERS([malloc.h malloc/malloc.h langinfo.h sys/param.h sys/mount.h sys/statvfs.h sys/select.h sockLib.h sys/mman.h sys/msg.h sys/vfs.h arpa/inet.h fcntl.h libintl.h netdb.h netinet/in.h netinet/in_systm.h netinet/ip.h sys/ioctl.h sys/socket.h sys/time.h unistd.h kstat.h sys/sysinfo.h kvm.h sys/file.h sys/resource.h ifaddrs.h mach/mach.h stddef.h sys/timeb.h terminos.h argz.h ucred.h endian.h sys/endian.h execinfo.h])
 
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 28689)
+++ src/Makefile.am	(working copy)
@@ -54,14 +54,10 @@
   lockmanager \
   dns \
   identity \
-  set \
   namestore \
-  vpn \
   gns \
   fs \
-  exit \
   pt \
-  integration-tests \
-  $(EXP_DIR) 
+  $(EXP_DIR)
 
 # note: namestore, gns are not listed right now as they are being reworked to use the new crypto
Index: src/dht/gnunet-service-dht_neighbours.c
===================================================================
--- src/dht/gnunet-service-dht_neighbours.c	(revision 28689)
+++ src/dht/gnunet-service-dht_neighbours.c	(working copy)
@@ -46,7 +46,6 @@
 #include "gnunet-service-dht_neighbours.h"
 #include "gnunet-service-dht_nse.h"
 #include "gnunet-service-dht_routing.h"
-#include <fenv.h>
 #include "dht.h"
 
 #define LOG_TRAFFIC(kind,...) GNUNET_log_from (kind, "dht-traffic",__VA_ARGS__)
Index: src/dns/Makefile.am
===================================================================
--- src/dns/Makefile.am	(revision 28689)
+++ src/dns/Makefile.am	(working copy)
@@ -17,14 +17,6 @@
 pkgcfg_DATA = \
   dns.conf
 
-if LINUX
-HIJACKBIN = gnunet-helper-dns
-install-exec-hook:
-	$(top_srcdir)/src/dns/install-dns-helper.sh $(libexecdir) $(GNUNETDNS_GROUP) $(SUDO_BINARY) || true
-else
-install-exec-hook:
-endif
-
 lib_LTLIBRARIES = \
   libgnunetdnsparser.la \
   libgnunetdnsstub.la \
Index: src/nat/Makefile.am
===================================================================
--- src/nat/Makefile.am	(revision 28689)
+++ src/nat/Makefile.am	(working copy)
@@ -14,16 +14,6 @@
 dist_pkgcfg_DATA = \
   nat.conf
 
-if LINUX
-  NATBIN = gnunet-helper-nat-server gnunet-helper-nat-client
-  NATSERVER = gnunet-helper-nat-server.c
-  NATCLIENT = gnunet-helper-nat-client.c
-install-exec-hook:
-	$(top_srcdir)/src/nat/install-nat-helper.sh $(libexecdir) $(SUDO_BINARY) || true
-else
-install-exec-hook:
-endif
-
 bin_PROGRAMS = \
  gnunet-nat-server 
 
@@ -97,4 +87,4 @@
 
 EXTRA_DIST = \
  test_nat_data.conf \
- test_nat_test_data.conf 
\ No newline at end of file
+ test_nat_test_data.conf 
Index: src/transport/Makefile.am
===================================================================
--- src/transport/Makefile.am	(revision 28689)
+++ src/transport/Makefile.am	(working copy)
@@ -46,20 +46,6 @@
 endif
 
 if LINUX
- WLAN_BIN = gnunet-helper-transport-wlan
- WLAN_BIN_DUMMY = gnunet-helper-transport-wlan-dummy
- WLAN_BIN_SENDER = gnunet-transport-wlan-sender
- WLAN_BIN_RECEIVER = gnunet-transport-wlan-receiver
- WLAN_PLUGIN_LA = libgnunet_plugin_transport_wlan.la
- WLAN_PLUGIN_TEST = test_plugin_wlan
- WLAN_API_TEST = test_transport_api_wlan
- WLAN_REL_TEST = test_transport_api_reliability_wlan
- WLAN_UREL_TEST = test_transport_api_unreliability_wlan
- WLAN_QUOTA_TEST = test_quota_compliance_wlan \
-		test_quota_compliance_wlan_asymmetric
-endif
-
-if LINUX
 install-exec-hook:
 	$(top_srcdir)/src/transport/install-wlan-helper.sh $(libexecdir) $(SUDO_BINARY) || true
 else
@@ -210,7 +196,6 @@
 
 plugin_LTLIBRARIES = \
   libgnunet_plugin_transport_tcp.la \
-  libgnunet_plugin_transport_udp.la \
   $(UNIX_PLUGIN_LA) \
   $(HTTP_CLIENT_PLUGIN_LA) \
   $(HTTPS_CLIENT_PLUGIN_LA) \
Index: src/util/Makefile.am
===================================================================
--- src/util/Makefile.am	(revision 28689)
+++ src/util/Makefile.am	(working copy)
@@ -97,12 +97,10 @@
   plugin.c \
   program.c \
   resolver_api.c resolver.h \
-  scheduler.c \
   server.c \
   server_mst.c \
   server_nc.c \
   server_tc.c \
-  service.c \
   signal.c \
   strings.c \
   time.c \
