Index: Makefile.am
===================================================================
--- Makefile.am	(revision 27966)
+++ Makefile.am	(working copy)
@@ -1,5 +1,5 @@
 INCLUDES = -I$(top_srcdir)/src/include
-SUBDIRS  = contrib doc m4 src po pkgconfig
+SUBDIRS  = libltdl contrib doc m4 src po pkgconfig
 
 EXTRA_DIST = \
  ABOUT-NLS \
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 27966)
+++ src/Makefile.am	(working copy)
@@ -47,9 +47,7 @@
   namestore \
   template \
   ats \
-  nat \
   fragmentation \
-  transport \
   ats-tool \
   peerinfo-tool \
   core \
@@ -63,8 +61,4 @@
   lockmanager \
   stream \
   dns \
-  fs \
-  $(LINUX_DIR) \
-  $(MINGW_DIR) \
-  integration-tests \
-  $(EXP_DIR) 
+  fs
Index: src/dht/gnunet-service-dht_neighbours.c
===================================================================
--- src/dht/gnunet-service-dht_neighbours.c	(revision 27966)
+++ src/dht/gnunet-service-dht_neighbours.c	(working copy)
@@ -46,7 +46,6 @@
 #include "gnunet-service-dht_neighbours.h"
 #include "gnunet-service-dht_nse.h"
 #include "gnunet-service-dht_routing.h"
-#include <fenv.h>
 #include "dht.h"
 
 #include "../regex/regex_block_lib.h"
Index: src/dns/Makefile.am
===================================================================
--- src/dns/Makefile.am	(revision 27966)
+++ src/dns/Makefile.am	(working copy)
@@ -17,14 +17,6 @@
 pkgcfg_DATA = \
   dns.conf
 
-if LINUX
-HIJACKBIN = gnunet-helper-dns
-install-exec-hook:
-	$(top_srcdir)/src/dns/install-dns-helper.sh $(libexecdir) $(GNUNETDNS_GROUP) $(SUDO_BINARY) || true
-else
-install-exec-hook:
-endif
-
 lib_LTLIBRARIES = \
   libgnunetdnsparser.la \
   libgnunetdnsstub.la \
Index: src/stream/stream_api.c
===================================================================
--- src/stream/stream_api.c	(revision 27966)
+++ src/stream/stream_api.c	(working copy)
@@ -2983,8 +2983,7 @@
     {
     case GNUNET_STREAM_OPTION_INITIAL_RETRANSMIT_TIMEOUT:
       /* Expect struct GNUNET_TIME_Relative */
-      socket->retransmit_timeout = va_arg (vargs,
-                                           struct GNUNET_TIME_Relative);
+      socket->retransmit_timeout.rel_value = va_arg (vargs, uint64_t);
       break;
     case GNUNET_STREAM_OPTION_TESTING_SET_WRITE_SEQUENCE_NUMBER:
       socket->testing_active = GNUNET_YES;
@@ -3264,8 +3263,7 @@
     switch (option)
     {
     case GNUNET_STREAM_OPTION_INITIAL_RETRANSMIT_TIMEOUT:
-      lsocket->retransmit_timeout = va_arg (vargs,
-                                            struct GNUNET_TIME_Relative);
+      lsocket->retransmit_timeout.rel_value = va_arg (vargs, uint64_t);
       break;
     case GNUNET_STREAM_OPTION_TESTING_SET_WRITE_SEQUENCE_NUMBER:
       lsocket->testing_active = GNUNET_YES;
