diff --git a/Makefile.am b/Makefile.am
index f1dbd09..516bece 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,8 +25,8 @@ DISTCHECK_CONFIGURE_FLAGS = --disable-random-daemon \
 # (A suitable gitlog-to-changelog script can be found in GnuPG master.)
 GITLOG_TO_CHANGELOG=gitlog-to-changelog
 
-DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS =         compat mpi cipher random src doc tests
+DIST_SUBDIRS = m4 compat mpi cipher random src
+SUBDIRS =         compat mpi cipher random src
 EXTRA_DIST = autogen.sh README.GIT 					\
              ChangeLog-2011 scripts/ChangeLog-2011 doc/ChangeLog-2011 	\
              m4/ChangeLog-2011 cipher/ChangeLog-2011 src/ChangeLog-2011 \
