diff --git a/configure.ac b/configure.ac
index 9088d02..08e9081 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1505,7 +1505,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS blowfish-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS blowfish-amd64.lo"
       ;;
       arm*-*-*)
          # Build with the assembly implementation
@@ -1522,7 +1522,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS cast5-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS cast5-amd64.lo"
       ;;
       arm*-*-*)
          # Build with the assembly implementation
@@ -1545,7 +1545,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS rijndael-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS rijndael-amd64.lo"
       ;;
       arm*-*-*)
          # Build with the assembly implementation
@@ -1562,7 +1562,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS twofish-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS twofish-amd64.lo"
       ;;
       arm*-*-*)
          # Build with the assembly implementation
@@ -1579,7 +1579,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the SSE2 implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS serpent-sse2-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS serpent-sse2-amd64.lo"
       ;;
    esac
 
@@ -1647,7 +1647,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_CIPHERS="$GCRYPT_CIPHERS salsa20-amd64.lo"
+         #GCRYPT_CIPHERS="$GCRYPT_CIPHERS salsa20-amd64.lo"
       ;;
    esac
 
@@ -1731,7 +1731,7 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha256-ssse3-amd64.lo"
+         #GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha256-ssse3-amd64.lo"
       ;;
    esac
 fi
@@ -1744,9 +1744,9 @@ if test "$found" = "1" ; then
    case "${host}" in
       x86_64-*-*)
          # Build with the assembly implementation
-         GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-ssse3-amd64.lo"
-         GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-avx-amd64.lo"
-         GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-avx2-bmi2-amd64.lo"
+         #GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-ssse3-amd64.lo"
+         #GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-avx-amd64.lo"
+         #GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha512-avx2-bmi2-amd64.lo"
       ;;
    esac
 
@@ -1776,7 +1776,7 @@ AC_DEFINE(USE_SHA1, 1,   [Defined if this module should be included])
 case "${host}" in
   x86_64-*-*)
     # Build with the assembly implementation
-    GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha1-ssse3-amd64.lo"
+    #GCRYPT_DIGESTS="$GCRYPT_DIGESTS sha1-ssse3-amd64.lo"
   ;;
 esac
 
diff --git a/random/random-csprng.c b/random/random-csprng.c
index 87235d8..92d8f81 100644
--- a/random/random-csprng.c
+++ b/random/random-csprng.c
@@ -1156,7 +1156,9 @@ getfnc_gather_random (void))(void (*)(const void*, size_t,
   return fnc;
 #endif
 
+#if 0 /* EMSCRIPTEN */
   log_fatal (_("no entropy gathering module detected\n"));
+#endif
 
   return NULL; /*NOTREACHED*/
 }
