Index: gnunet-svn-32653/configure.ac
===================================================================
--- gnunet-svn-32653/configure.ac	(revision 32653)
+++ gnunet-svn-32653/configure.ac	(working copy)
@@ -582,7 +582,7 @@
     AC_CHECK_HEADERS(ltdl.h,
      AC_CHECK_LIB([ltdl], [lt_dlopenext],
       ltdl=1))])
-if test x$ltdl = x1
+if true
 then
  AC_MSG_RESULT([libltdl found])
 else
Index: gnunet-svn-32653/src/Makefile.am
===================================================================
--- gnunet-svn-32653/src/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/Makefile.am	(working copy)
@@ -64,10 +64,8 @@
   peerinfo-tool \
   core \
   $(TESTBED) \
-  ats-tests \
   nse \
   dht \
-  hostlist \
   topology \
   regex \
   mesh \
@@ -78,11 +76,6 @@
   gnsrecord \
   namecache \
   namestore \
-  vpn \
-  gns \
   $(CONVERSATION_DIR) \
   fs \
-  exit \
-  pt \
-  integration-tests \
   $(EXP_DIR)
Index: gnunet-svn-32653/src/arm/gnunet-service-arm.c
===================================================================
--- gnunet-svn-32653/src/arm/gnunet-service-arm.c	(revision 32653)
+++ gnunet-svn-32653/src/arm/gnunet-service-arm.c	(working copy)
@@ -1561,19 +1561,5 @@
 }
 
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor)) GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
-
-
 /* end of gnunet-service-arm.c */
Index: gnunet-svn-32653/src/dht/gnunet-service-dht_neighbours.c
===================================================================
--- gnunet-svn-32653/src/dht/gnunet-service-dht_neighbours.c	(revision 32653)
+++ gnunet-svn-32653/src/dht/gnunet-service-dht_neighbours.c	(working copy)
@@ -46,7 +46,6 @@
 #include "gnunet-service-dht_neighbours.h"
 #include "gnunet-service-dht_nse.h"
 #include "gnunet-service-dht_routing.h"
-#include <fenv.h>
 #include "dht.h"
 
 #define LOG_TRAFFIC(kind,...) GNUNET_log_from (kind, "dht-traffic",__VA_ARGS__)
Index: gnunet-svn-32653/src/dns/Makefile.am
===================================================================
--- gnunet-svn-32653/src/dns/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/dns/Makefile.am	(working copy)
@@ -17,14 +17,6 @@
 pkgcfg_DATA = \
   dns.conf
 
-if LINUX
-HIJACKBIN = gnunet-helper-dns
-install-exec-hook:
-	$(top_srcdir)/src/dns/install-dns-helper.sh $(libexecdir) $(GNUNETDNS_GROUP) $(SUDO_BINARY) || true
-else
-install-exec-hook:
-endif
-
 lib_LTLIBRARIES = \
   libgnunetdnsparser.la \
   libgnunetdnsstub.la \
Index: gnunet-svn-32653/src/nat/Makefile.am
===================================================================
--- gnunet-svn-32653/src/nat/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/nat/Makefile.am	(working copy)
@@ -14,24 +14,6 @@
 dist_pkgcfg_DATA = \
   nat.conf
 
-if LINUX
-  NATBIN = gnunet-helper-nat-server gnunet-helper-nat-client
-  NATSERVER = gnunet-helper-nat-server.c
-  NATCLIENT = gnunet-helper-nat-client.c
-install-exec-hook:
-	$(top_srcdir)/src/nat/install-nat-helper.sh $(libexecdir) $(SUDO_BINARY) || true
-else
-if XFREEBSD
-  NATBIN = gnunet-helper-nat-server gnunet-helper-nat-client
-  NATSERVER = gnunet-helper-nat-server.c
-  NATCLIENT = gnunet-helper-nat-client.c
-install-exec-hook:
-	$(top_srcdir)/src/nat/install-nat-helper.sh $(libexecdir) $(SUDO_BINARY) || true
-endif
-else
-install-exec-hook:
-endif
-
 bin_PROGRAMS = \
  gnunet-nat-server 
 
Index: gnunet-svn-32653/src/nse/gnunet-service-nse.c
===================================================================
--- gnunet-svn-32653/src/nse/gnunet-service-nse.c	(revision 32653)
+++ gnunet-svn-32653/src/nse/gnunet-service-nse.c	(working copy)
@@ -1539,21 +1539,7 @@
 }
 
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor))
-GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
 
 
-
 /* end of gnunet-service-nse.c */
Index: gnunet-svn-32653/src/regex/gnunet-daemon-regexprofiler.c
===================================================================
--- gnunet-svn-32653/src/regex/gnunet-daemon-regexprofiler.c	(revision 32653)
+++ gnunet-svn-32653/src/regex/gnunet-daemon-regexprofiler.c	(working copy)
@@ -382,19 +382,6 @@
 }
 
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor)) GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
 
-
 /* end of gnunet-daemon-regexprofiler.c */
Index: gnunet-svn-32653/src/revocation/gnunet-service-revocation.c
===================================================================
--- gnunet-svn-32653/src/revocation/gnunet-service-revocation.c	(revision 32653)
+++ gnunet-svn-32653/src/revocation/gnunet-service-revocation.c	(working copy)
@@ -889,21 +889,7 @@
 }
 
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor))
-GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
 
 
-
 /* end of gnunet-service-revocation.c */
Index: gnunet-svn-32653/src/set/Makefile.am
===================================================================
--- gnunet-svn-32653/src/set/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/set/Makefile.am	(working copy)
@@ -24,24 +24,6 @@
 lib_LTLIBRARIES = \
   libgnunetset.la
 
-gnunet_set_profiler_SOURCES = \
- gnunet-set-profiler.c
-gnunet_set_profiler_LDADD = \
-  $(top_builddir)/src/util/libgnunetutil.la \
-  $(top_builddir)/src/set/libgnunetset.la \
-  $(top_builddir)/src/testing/libgnunettesting.la \
-  $(GN_LIBINTL)
-gnunet_set_profiler_DEPENDENCIES = \
-  libgnunetset.la
-
-
-gnunet_set_ibf_profiler_SOURCES = \
- gnunet-set-ibf-profiler.c \
- ibf.c
-gnunet_set_ibf_profiler_LDADD = \
-  $(top_builddir)/src/util/libgnunetutil.la \
-  $(GN_LIBINTL)
-
 gnunet_service_set_SOURCES = \
  gnunet-service-set.c gnunet-service-set.h \
  gnunet-service-set_union.c \
Index: gnunet-svn-32653/src/statistics/gnunet-service-statistics.c
===================================================================
--- gnunet-svn-32653/src/statistics/gnunet-service-statistics.c	(revision 32653)
+++ gnunet-svn-32653/src/statistics/gnunet-service-statistics.c	(working copy)
@@ -861,19 +861,6 @@
                               GNUNET_SERVICE_OPTION_SOFT_SHUTDOWN, &run, NULL)) ? 0 : 1;
 }
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor)) GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
 
-
 /* end of gnunet-service-statistics.c */
Index: gnunet-svn-32653/src/topology/gnunet-daemon-topology.c
===================================================================
--- gnunet-svn-32653/src/topology/gnunet-daemon-topology.c	(revision 32653)
+++ gnunet-svn-32653/src/topology/gnunet-daemon-topology.c	(working copy)
@@ -1294,18 +1294,5 @@
 }
 
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor)) GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
-
 /* end of gnunet-daemon-topology.c */
Index: gnunet-svn-32653/src/transport/Makefile.am
===================================================================
--- gnunet-svn-32653/src/transport/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/transport/Makefile.am	(working copy)
@@ -46,21 +46,6 @@
 endif
 
 if LINUX
- WLAN_BIN = gnunet-helper-transport-wlan
- WLAN_BIN_DUMMY = gnunet-helper-transport-wlan-dummy
- WLAN_BIN_SENDER = gnunet-transport-wlan-sender
- WLAN_BIN_RECEIVER = gnunet-transport-wlan-receiver
- WLAN_PLUGIN_LA = libgnunet_plugin_transport_wlan.la
- WLAN_PLUGIN_TEST = test_plugin_wlan
- WLAN_API_TEST = test_transport_api_wlan
- WLAN_TIMEOUT_TEST = test_transport_api_timeout_wlan
- WLAN_REL_TEST = test_transport_api_reliability_wlan
- WLAN_UREL_TEST = test_transport_api_unreliability_wlan
- WLAN_QUOTA_TEST = test_quota_compliance_wlan \
-		test_quota_compliance_wlan_asymmetric
-endif
-
-if LINUX
 install-exec-hook:
 	$(top_srcdir)/src/transport/install-wlan-helper.sh $(libexecdir) $(SUDO_BINARY) || true
 else
@@ -233,18 +218,6 @@
   $(CFLAGS)
 # -DANALYZE
 
-plugin_LTLIBRARIES = \
-  libgnunet_plugin_transport_tcp.la \
-  libgnunet_plugin_transport_udp.la \
-  $(UNIX_PLUGIN_LA) \
-  $(HTTP_CLIENT_PLUGIN_LA) \
-  $(HTTPS_CLIENT_PLUGIN_LA) \
-  $(HTTP_SERVER_PLUGIN_LA) \
-  $(HTTPS_SERVER_PLUGIN_LA) \
-  $(WLAN_PLUGIN_LA) \
-  $(BT_PLUGIN_LA) \
-  libgnunet_plugin_transport_template.la
-
 libgnunet_plugin_transport_tcp_la_SOURCES = \
   plugin_transport_tcp.c
 libgnunet_plugin_transport_tcp_la_LIBADD = \
Index: gnunet-svn-32653/src/util/Makefile.am
===================================================================
--- gnunet-svn-32653/src/util/Makefile.am	(revision 32653)
+++ gnunet-svn-32653/src/util/Makefile.am	(working copy)
@@ -72,12 +72,9 @@
 libgnunetutil_la_SOURCES = \
   bandwidth.c \
   bio.c \
-  client.c \
   common_allocation.c \
   common_endian.c \
   common_logging.c \
-  configuration.c \
-  connection.c \
   container_bloomfilter.c \
   container_heap.c \
   container_meta_data.c \
@@ -106,15 +103,10 @@
   os_network.c \
   os_priority.c \
   peer.c \
-  plugin.c \
-  program.c \
   resolver_api.c resolver.h \
-  scheduler.c \
-  server.c \
   server_mst.c \
   server_nc.c \
   server_tc.c \
-  service.c \
   signal.c \
   strings.c \
   time.c \
Index: gnunet-svn-32653/src/util/gnunet-service-resolver.c
===================================================================
--- gnunet-svn-32653/src/util/gnunet-service-resolver.c	(revision 32653)
+++ gnunet-svn-32653/src/util/gnunet-service-resolver.c	(working copy)
@@ -590,19 +590,6 @@
   return ret;
 }
 
-#ifdef LINUX
-#include <malloc.h>
 
-/**
- * MINIMIZE heap size (way below 128k) since this process doesn't need much.
- */
-void __attribute__ ((constructor)) GNUNET_ARM_memory_init ()
-{
-  mallopt (M_TRIM_THRESHOLD, 4 * 1024);
-  mallopt (M_TOP_PAD, 1 * 1024);
-  malloc_trim (0);
-}
-#endif
 
-
 /* end of gnunet-service-resolver.c */
