Index: gnunet-svn-35965/configure.ac
===================================================================
--- gnunet-svn-35965/configure.ac	(revision 35965)
+++ gnunet-svn-35965/configure.ac	(working copy)
@@ -621,7 +621,7 @@
     AC_CHECK_HEADERS(ltdl.h,
      AC_CHECK_LIB([ltdl], [lt_dlopenext],
       ltdl=1))])
-if test x$ltdl = x1
+if true
 then
  AC_MSG_RESULT([libltdl found])
 else
Index: gnunet-svn-35965/src/Makefile.am
===================================================================
--- gnunet-svn-35965/src/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/Makefile.am	(working copy)
@@ -77,10 +77,8 @@
   peerinfo-tool \
   core \
   $(TESTBED) \
-  ats-tests \
   nse \
   dht \
-  hostlist \
   topology \
   regex \
   cadet \
@@ -92,14 +90,9 @@
   gnsrecord \
   namecache \
   namestore \
-  vpn \
-  gns \
   $(CONVERSATION_DIR) \
   peerstore \
   fs \
-  exit \
-  pt \
-  integration-tests \
   $(EXP_DIR)
 
 endif
Index: gnunet-svn-35965/src/datastore/Makefile.am
===================================================================
--- gnunet-svn-35965/src/datastore/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/datastore/Makefile.am	(working copy)
@@ -104,7 +104,8 @@
   $(SQLITE_PLUGIN) \
   $(MYSQL_PLUGIN) \
   $(POSTGRES_PLUGIN) \
-  libgnunet_plugin_datastore_heap.la
+  libgnunet_plugin_datastore_heap.la \
+  libgnunet_plugin_datastore_emscripten.la
 
 # Real plugins should of course go into
 # plugin_LTLIBRARIES
@@ -130,6 +131,13 @@
 libgnunet_plugin_datastore_heap_la_LDFLAGS = \
  $(GN_PLUGIN_LDFLAGS)
 
+libgnunet_plugin_datastore_emscripten_la_SOURCES = \
+  plugin_datastore_emscripten.c
+libgnunet_plugin_datastore_emscripten_la_LIBADD = \
+  $(top_builddir)/src/util/libgnunetutil.la $(XLIBS) \
+  $(LTLIBINTL)
+libgnunet_plugin_datastore_emscripten_la_LDFLAGS = \
+ $(GN_PLUGIN_LDFLAGS)
 
 libgnunet_plugin_datastore_mysql_la_SOURCES = \
   plugin_datastore_mysql.c
Index: gnunet-svn-35965/src/dht/Makefile.am
===================================================================
--- gnunet-svn-35965/src/dht/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/dht/Makefile.am	(working copy)
@@ -55,12 +55,6 @@
  gnunet-service-dht-whanau
 endif
 
-noinst_PROGRAMS = \
- gnunet-dht-monitor \
- gnunet-dht-get \
- gnunet-dht-put \
- gnunet-dht-profiler
-
 gnunet_service_dht_SOURCES = \
  gnunet-service-dht.c gnunet-service-dht.h \
  gnunet-service-dht_clients.c gnunet-service-dht_clients.h \
Index: gnunet-svn-35965/src/dns/Makefile.am
===================================================================
--- gnunet-svn-35965/src/dns/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/dns/Makefile.am	(working copy)
@@ -18,14 +18,6 @@
 pkgcfg_DATA = \
   dns.conf
 
-if LINUX
-HIJACKBIN = gnunet-helper-dns
-install-exec-hook:
-	$(top_srcdir)/src/dns/install-dns-helper.sh $(libexecdir) $(GNUNETDNS_GROUP) $(SUDO_BINARY) || true
-else
-install-exec-hook:
-endif
-
 lib_LTLIBRARIES = \
   libgnunetdnsparser.la \
   libgnunetdnsstub.la \
Index: gnunet-svn-35965/src/nat/Makefile.am
===================================================================
--- gnunet-svn-35965/src/nat/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/nat/Makefile.am	(working copy)
@@ -15,24 +15,6 @@
 dist_pkgcfg_DATA = \
   nat.conf
 
-if LINUX
-  NATBIN = gnunet-helper-nat-server gnunet-helper-nat-client
-  NATSERVER = gnunet-helper-nat-server.c
-  NATCLIENT = gnunet-helper-nat-client.c
-install-exec-hook:
-	$(top_srcdir)/src/nat/install-nat-helper.sh $(libexecdir) $(SUDO_BINARY) || true
-else
-if XFREEBSD
-  NATBIN = gnunet-helper-nat-server gnunet-helper-nat-client
-  NATSERVER = gnunet-helper-nat-server.c
-  NATCLIENT = gnunet-helper-nat-client.c
-install-exec-hook:
-	$(top_srcdir)/src/nat/install-nat-helper.sh $(libexecdir) $(SUDO_BINARY) || true
-endif
-else
-install-exec-hook:
-endif
-
 bin_PROGRAMS = \
  gnunet-nat-server 
 
Index: gnunet-svn-35965/src/peerstore/Makefile.am
===================================================================
--- gnunet-svn-35965/src/peerstore/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/peerstore/Makefile.am	(working copy)
@@ -52,7 +52,7 @@
   $(GN_LIB_LDFLAGS)
 
 plugin_LTLIBRARIES = \
-  libgnunet_plugin_peerstore_sqlite.la
+  libgnunet_plugin_peerstore_emscripten.la
 
 libgnunet_plugin_peerstore_sqlite_la_SOURCES = \
   plugin_peerstore_sqlite.c
@@ -63,6 +63,18 @@
 libgnunet_plugin_peerstore_sqlite_la_LDFLAGS = \
  $(GN_PLUGIN_LDFLAGS)
 
+libgnunet_plugin_peerstore_emscripten_la_SOURCES = \
+  plugin_peerstore_emscripten.c
+libgnunet_plugin_peerstore_emscripten_la_LIBADD = \
+  $(top_builddir)/src/peerstore/libgnunetpeerstore.la  \
+  $(top_builddir)/src/util/libgnunetutil.la \
+  $(LTLIBINTL)
+libgnunet_plugin_peerstore_emscripten_la_LDFLAGS = \
+ $(GN_PLUGIN_LDFLAGS)
+libgnunet_plugin_peerstore_emscripten_la_DEPENDENCIES = \
+  $(top_builddir)/src/util/libgnunetutil.la \
+  libgnunetpeerstore.la
+
 check_PROGRAMS = \
  test_peerstore_api_store \
  test_peerstore_api_iterate \
Index: gnunet-svn-35965/src/set/Makefile.am
===================================================================
--- gnunet-svn-35965/src/set/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/set/Makefile.am	(working copy)
@@ -16,9 +16,6 @@
   AM_CFLAGS = -fprofile-arcs -ftest-coverage
 endif
 
-noinst_PROGRAMS = \
- gnunet-set-profiler gnunet-set-ibf-profiler
-
 libexec_PROGRAMS = \
  gnunet-service-set
 
Index: gnunet-svn-35965/src/transport/Makefile.am
===================================================================
--- gnunet-svn-35965/src/transport/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/transport/Makefile.am	(working copy)
@@ -80,20 +80,6 @@
 endif
 
 if LINUX
- WLAN_BIN = gnunet-helper-transport-wlan
- WLAN_BIN_DUMMY = gnunet-helper-transport-wlan-dummy
- WLAN_BIN_SENDER = gnunet-transport-wlan-sender
- WLAN_BIN_RECEIVER = gnunet-transport-wlan-receiver
- WLAN_PLUGIN_LA = libgnunet_plugin_transport_wlan.la
- WLAN_PLUGIN_TEST = test_plugin_wlan
- WLAN_API_TEST = test_transport_api_wlan
- WLAN_TIMEOUT_TEST = test_transport_api_timeout_wlan
- WLAN_REL_TEST = test_transport_api_reliability_wlan
- WLAN_QUOTA_TEST = test_quota_compliance_wlan \
-		test_quota_compliance_wlan_asymmetric
-endif
-
-if LINUX
 install-exec-hook:
 	$(top_srcdir)/src/transport/install-wlan-helper.sh $(libexecdir) $(SUDO_BINARY) || true
 if HAVE_LIBBLUETOOTH
@@ -269,22 +255,8 @@
   $(CFLAGS)
 # -DANALYZE
 
-plugin_LTLIBRARIES = \
-  libgnunet_plugin_transport_tcp.la \
-  libgnunet_plugin_transport_udp.la \
-  $(UNIX_PLUGIN_LA) \
-  $(HTTP_CLIENT_PLUGIN_LA) \
-  $(HTTPS_CLIENT_PLUGIN_LA) \
-  $(HTTP_SERVER_PLUGIN_LA) \
-  $(HTTPS_SERVER_PLUGIN_LA) \
-  $(WLAN_PLUGIN_LA) \
-  $(BT_PLUGIN_LA)
+plugin_LTLIBRARIES = libgnunet_plugin_transport_http_client.la
 
-# Note: real plugins of course need to be added
-# to the plugin_LTLIBRARIES above
-noinst_LTLIBRARIES = \
-  libgnunet_plugin_transport_template.la
-
 libgnunet_plugin_transport_tcp_la_SOURCES = \
   plugin_transport_tcp.c
 libgnunet_plugin_transport_tcp_la_LIBADD = \
@@ -358,7 +330,7 @@
 
 
 libgnunet_plugin_transport_http_client_la_SOURCES = \
-  plugin_transport_http_client.c plugin_transport_http_common.c plugin_transport_http_common.h
+  plugin_transport_http_client_emscripten.c plugin_transport_http_common.c plugin_transport_http_common.h
 libgnunet_plugin_transport_http_client_la_LIBADD = \
   $(top_builddir)/src/hello/libgnunethello.la \
   $(top_builddir)/src/statistics/libgnunetstatistics.la \
Index: gnunet-svn-35965/src/util/Makefile.am
===================================================================
--- gnunet-svn-35965/src/util/Makefile.am	(revision 35965)
+++ gnunet-svn-35965/src/util/Makefile.am	(working copy)
@@ -59,14 +59,10 @@
 libgnunetutil_la_SOURCES = \
   bandwidth.c \
   bio.c \
-  client.c \
   client_manager.c \
   common_allocation.c \
   common_endian.c \
   common_logging.c \
-  configuration.c \
-  configuration_loader.c \
-  connection.c \
   container_bloomfilter.c \
   container_heap.c \
   container_meta_data.c \
@@ -98,15 +94,11 @@
   os_network.c \
   os_priority.c \
   peer.c \
-  plugin.c \
-  program.c \
   resolver_api.c resolver.h \
   scheduler.c \
-  server.c \
   server_mst.c \
   server_nc.c \
   server_tc.c \
-  service.c \
   signal.c \
   strings.c \
   time.c \
