commit e1ba13fb602b82c80ead2cc0bafc9341993b3f53
Author: David Barksdale <amatus@amatus.name>
Date:   Thu Jul 3 18:09:52 2014 -0500

    Fix IDBFS for webworkers

diff --git a/src/library_idbfs.js b/src/library_idbfs.js
index 8082c19..d9718e8 100644
--- a/src/library_idbfs.js
+++ b/src/library_idbfs.js
@@ -3,7 +3,7 @@ mergeInto(LibraryManager.library, {
   $IDBFS: {
     dbs: {},
     indexedDB: function() {
-      return window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
+      return indexedDB || window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
     },
     DB_VERSION: 21,
     DB_STORE_NAME: 'FILE_DATA',
